keyring/credentials.py

```diff
@@ -39,6 +39,15 @@ def __init__(self, user_env_var, pwd_env_var):
         self.user_env_var = user_env_var
         self.pwd_env_var = pwd_env_var
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, EnvironCredential):
+            return NotImplemented
+
+        return (
+            self.user_env_var == other.user_env_var
+            and self.pwd_env_var == other.pwd_env_var
+        )
```

Comment:
I think this can be more concisely represented by:

```suggestion
        return vars(self) == vars(other)
```

I don't think it's necessary to return NotImplemented, as Python simply transforms that into False anyway, and it seems a vanishingly small chance that anything other than an EnvironCredential would have the same internal state and end up being compared with an EnvironCredential for equality.